components/director/internal/domain/scenarioassignment/repository.go

```diff
@@ -127,3 +137,16 @@ func (r *repository) DeleteForScenarioName(ctx context.Context, tenantID string,
 
 	return r.deleter.DeleteOne(ctx, tenantID, conditions)
 }
+
+func (r *repository) EnsureScenarioAssigned(ctx context.Context, in model.AutomaticScenarioAssignment) error {
```

Comment:
I tested it and it doesn't work right way. Basically it removes all other scenarios for given runtimes. It should preserve them and just append a new one from ASA.

- Add new scenario
``` graphql
mutation addNewScenario {
  createLabelDefinition(in: {
    key:"scenarios",
    schema: "{\"items\":{\"enum\":[\"DEFAULT\", \"FOO\", \"BAR\"],\"maxLength\":128,\"pattern\":\"^[A-Za-z0-9]([-_A-Za-z0-9\\\\s]*[A-Za-z0-9])$\",\"type\":\"string\"},\"minItems\":1,\"type\":\"array\",\"uniqueItems\":true}"  }) {
    key
      schema
  }
}
```
- Create runtimes

```graphql

mutation CreateRTM1 {
  registerRuntime(in:{
    name:"foo",
    labels: {
      foo:"bar",
      scenarios: ["DEFAULT", "FOO"],
    }
  }) {
    id
    name
  }
}

mutation CreateRTM2 {
  registerRuntime(in:{
    name:"bar",
    labels: {
      foo:"bar",
      scenarios: ["DEFAULT"],
    }
  }) {
    id
    name
  }
}

mutation CreateRTM3 {
  registerRuntime(in:{
    name:"baz",
    labels: {
      foo:"bar",
      scenarios: ["DEFAULT", "BAR"],
    }
  }) {
    id
    name
  }
}
```

- Create ASA

```graphql
mutation CreateASA {
  createAutomaticScenarioAssignment(in:{
    scenarioName:"FOO",
    selector:{
      key:"foo",
      value:"bar"
    }
  }) {
    scenarioName
    selector {
      key
      value
    }
  }
}
```

- List Runtimes:
```graphql
  runtimes {
    data {
      id
      name
      labels
    }
  }
```

Observe that both runtimes have just `FOO` scenario assigned:

```json
  {
  "data": {
    "runtimes": {
      "data": [
        {
          "id": "f1fec278-37bb-4d16-a61b-611968e46c09",
          "name": "bar",
          "labels": {
            "foo": "bar",
            "scenarios": [
              "FOO"
            ]
          }
        },
        {
          "id": "63b35c2c-65c8-49e8-91cf-f167a0f72447",
          "name": "baz",
          "labels": {
            "foo": "bar",
            "scenarios": [
              "FOO"
            ]
          }
        },
        {
          "id": "e91385d5-59a9-46a0-8e30-c11f2abaa09e",
          "name": "foo",
          "labels": {
            "foo": "bar",
            "scenarios": [
              "FOO"
            ]
          }
        }
      ]
    }
  }
}
```